src/charm.py

```diff
@@ -3,61 +3,77 @@
 import sys
 sys.path.append('lib')
 
-from ops.charm import CharmBase
-from ops.framework import StoredState
+from ops.charm import CharmBase, CharmEvents
+from ops.framework import EventBase, EventSource, StoredState
 from ops.main import main
 from ops.model import (
     ActiveStatus,
-    BlockedStatus,
-    WaitingStatus,
     MaintenanceStatus,
+    WaitingStatus,
 )
 
-from interfaces import (
-    HTTPInterfaceProvides,
-    MySQLInterfaceRequires,
-)
-from resources import OCIImageResource
+from interface_http import HTTPServer
+from interface_mysql import MySQLServer, MySQLServerError
+from oci_image import OCIImageResource, FetchError
+
+
+class ConfigurePodEvent(EventBase):
+    pass
+
+
+class ServeWebsiteEvent(EventBase):
+    pass
+
+
+class GitLabK8sCharmEvents(CharmEvents):
```

Comment:
No need to have K8s on every single type related to k8s I think?  GitLabCharm is shorter and sweeter to read.